### DISCUSSION TITLE
Turn any page into an API

### DISCUSSION THREAD

**Discussion by atej** [johnlindquist/kit#672] [Share]
Find yourself visiting a page often just to see what's new?

The `list` helper gets you the latest posts right within the Kit app.

### Install
In your `~/.kenv`: `npm install @aseemtaneja/kit-list`

### API-fy a page
Step 1: Go to the page  
Step 2: Open up the Elements tab in Dev Tools and gather the required CSS selectors  
Step 3: Make a script and you're done  

For example, the following script allows you to browse Frontend Masters courses:
```javascript
// Name: Frontend Masters
// Description: Browse courses from frontendmasters.com
// Author: Aseem Taneja
// Twitter: @aseemtaneja

import '@johnlindquist/kit';
const list = await npm('@aseemtaneja/kit-list');

const { choices } = await list('https://frontendmasters.com/courses/', {
        containerSelector: '.MediaItem',
        hrefSelector: 'h2 a',
        descriptionSelector: '.description',
        metaSelector: '.Instructor .name',
      });

const itemUrl = await arg('Go to', choices);

await $`open ${itemUrl}`;
```

To see everything you can do  👉🏽 [View the docs on GitHub](https://github.com/atej/kit-list/)

**Comment by johnlindquist** [comment]
This looks great!

Would you mind adding a little bit more info to your post? It would really help the generated pages look much nicer:

https://www.scriptkit.com/atej/a-list-helper-to-consume-regular-html-pages-like-apis
<img width="796" alt="CleanShot 2021-12-27 at 13 07 31@2x" src="https://user-images.githubusercontent.com/36073/147503979-b589209b-4ea2-4cea-b362-858cfae55a33.png">
<img width="814" alt="CleanShot 2021-12-27 at 13 07 43@2x" src="https://user-images.githubusercontent.com/36073/147503982-b928ff53-1a55-4f57-89c9-52541a05a500.png">

**Reply by atej** [reply]
Updated the description. Thank you!

**Comment by cfritze64** [comment]
Cool-o. Initial test adapting it to 9to5mac.com is working great.
